typings/pluginapi.d.ts

```diff
@@ -78,6 +82,75 @@ import * as express from "express"
  * ]
  */
 
+export enum HttpCode {
+  Ok = 200,
+  Redirect = 302,
+  NotFound = 404,
+  BadRequest = 400,
+  Unauthorized = 401,
+  LargePayload = 413,
+  ServerError = 500,
+}
+
+export declare class HttpError extends Error {
+  constructor(message: string, status: HttpCode, details?: object)
+}
+
+export interface WebsocketRequest extends express.Request {
+  ws: net.Socket
+  head: Buffer
+}
+
+export type WebSocketHandler = (
+  req: WebsocketRequest,
+  res: express.Response,
+  next: express.NextFunction,
+) => void | Promise<void>
+
+export interface WebsocketRouter {
+  readonly router: express.Router
+  ws(route: expressCore.PathParams, ...handlers: WebSocketHandler[]): void
+}
+
+/**
+ * Create a router for websocket routes.
+ */
+export function WsRouter(): WebsocketRouter
+
+/**
+ * The websocket server used by code-server.
+ */
+export const wss: Websocket.Server
+
+/**
+ * The Express import used by code-server.
+ *
+ * Re-exported so plugins don't have to import duplicate copies of Express and
+ * to avoid potential version differences or issues caused by running separate
+ * instances.
+ */
+export { express }
+/**
+ * Use to add a field to a log.
+ *
+ * Re-exported so plugins don't have to import duplicate copies of the logger.
+ */
+export { field, Level, Logger }
```

Comment:
Or a util module?

Reply:
I'm down to organize this differently. The only upshot to leaving it that I can think of is that the usage is a bit shorter, i.e. `cs.field` instead of `cs.util.field` if you have `import * as cs from "code-server"`.

Oh definitely down to do something like `import * as util from "code-server/util"`, is that what you mean?

Reply:
I suppose having it all on the one import might be more ergonomic though? In which case I'm down for the `util` object or namespace.

Reply:
Is there any large reason (performance or otherwise) to move it? 

Otherwise, probably makes for a nice API for the plugin authors calling `cs.field` etc.